docs/contributor/02-10-testing-strategy.md

```diff
@@ -0,0 +1,65 @@
+# Testing strategy of the Istio module
```

Comment:
I think we should add an additional section describing what we want to test, like the main areas/categories of tests, for example:
- Istio installation and upgrade
- Istio functionalities
- Gateway API, etc.